torch/csrc/autograd/cpp_hook.cpp

```diff
@@ -14,15 +14,15 @@ void check_single_result (Variable value, Variable result, std::string hook_name
 
 namespace torch { namespace autograd {
 
-CppFunctionPreHook::CppFunctionPreHook(const std::shared_ptr<hooks_list> &hooks, int value_idx)
-: hooks_(hooks)
+CppFunctionPreHook::CppFunctionPreHook(std::shared_ptr<CppHooksList> hooks, int value_idx)
```

Comment:
Question for personal curiosity: why remove the const & and add a move here?

Reply:
CppFunctionPreHook always retains a copy of CppHooksList. In this situation, pass-by-value is superior to pass-by-reference, in the case that you call this constructor with a temporary. With pass-by-reference, you have to make a copy of the temporary; here, the temporary can be directly moved into the function.